common.blocks/i-bem-dom/__events/i-bem-dom__events.js

```diff
@@ -52,38 +52,44 @@ var undef,
          * @returns {EventManager} this
          */
         on : function(e, data, fn, _fnCtx, _isOnce) {
-            var params = this._params,
-                event = this._eventBuilder(e, params, 'on');
-
             if(functions.isFunction(data)) {
                 _isOnce = _fnCtx;
                 _fnCtx = fn;
                 fn = data;
                 data = undef;
             }
 
-            var fnStorage = this._storage[event] || (this._storage[event] = {}),
-                fnId = identify(fn, _fnCtx);
-
-            if(!fnStorage[fnId]) {
-                var bindDomNodes = params.bindDomNodes,
-                    bindClassName = params.bindClassName,
-                    _this = this,
-                    handler = fnStorage[fnId] = this._handlerWrapper(
-                        _isOnce?
-                            function() {
-                                _this.un(e, fn, _fnCtx);
-                                fn.apply(this, arguments);
-                            } :
-                            fn,
-                        data,
-                        _fnCtx,
-                        fnId);
+            var _this = this,
+                params = this._params,
+                bindDomNodes = params.bindDomNodes,
+                bindClassName = params.bindClassName,
+                fnId = identify(fn, _fnCtx),
+                onEvent = function(eventName) {
+                    var event = this._eventBuilder(eventName, params, 'on'),
+                        fnStorage = this._storage[event] || (this._storage[event] = {});
+
+                    if(!fnStorage[fnId]) {
+                        fnStorage[fnId] = handler? handler : handler = this._handlerWrapper(
+                            _isOnce?
+                                function() {
+                                    _this.un(e, fn, _fnCtx);
+                                    fn.apply(this, arguments);
+                                } :
+                                fn,
+                            data,
+                            _fnCtx,
+                            fnId);
+
+                        bindDomNodes.forEach(function(domNode) {
+                            domNode.addEventListener(event, handler, false);
+                        });
+                    }
+                }.bind(this),
+                handler;
 
-                bindDomNodes.forEach(function(domNode) {
-                    domNode.addEventListener(event, handler, false);
-                });
-            }
+            typeof e === 'string'?
```

Comment:
it's better to do this in very first lines of this function — just check if `e` is a string and contains spaces and do `forEach` of `this.on` (we do similar https://github.com/bem/bem-core/blob/v4/common.blocks/i-bem-dom/__events/i-bem-dom__events.js#L226)